### PR TITLE
Add clear button

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR addresses #14 by adding a button to clear the workflow.

Full disclaimer this is my **first ever** React PR so any feedback is appreciated 🤓

### CODE REVIEW

File: src/WorkflowClear.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const WorkflowClear = (): JSX.Element => {
+  const { clearNodes } = useWorkflow()
+
+  function clearWorkflow (): void {
+    clearNodes()
+  }
+
+  return (
+    <button className='btn btn-outline-danger' onClick={clearWorkflow}>
```
Comment (sverhoeven) [inline src/WorkflowClear.tsx:11]: The wrapping of clearNodes is not needed. You can directly use clearNodes as onClick value.
```suggestion
  return (
    <button className='btn btn-outline-danger' onClick={clearNodes}>
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Great first react contribution.

I expected that when you load the example and press the clear button then in the text tab the text and file list would be empty, but the text still shows the global parameters and the file list shows files that where previously linked in the nodes.

Could you besides clearing the nodes array also clear the global and files objects? 

Clearing can be done to setting them to `{}`. Typescript will complain about empty object not having the correct type so you will need to use a typed assignment or cast

**Comment by rvhonorato** [comment]
> Clearing can be done to setting them to `{}`. Typescript will complain about empty object not having the correct type so you will need to use a typed assignment or cast

Thanks for the tip! 

It should now be clearing the globals and the files, please check if the implementation is ok

File: src/store.ts [hunk 2]
```diff
@@ -167,6 +168,12 @@ export function useWorkflow (): UseWorkflow {
       setGlobal(parameters)
       setFiles(newUsedFiles)
     },
+    clearNodes () {
+      const blankNodes = removeAllItems(nodes)
+      const blankGlobals = emptyParams()
+      setNodes(blankNodes)
+      setGlobal(blankGlobals)
```
Comment (sverhoeven) [inline src/store.ts:175]: The files are not cleared yet
```suggestion
      setGlobal(blankGlobals)
      const blankFiles: IFiles = {}
      setFiles(blankFiles)
```

File: src/store.ts [hunk 3]
```diff
@@ -167,6 +168,12 @@ export function useWorkflow (): UseWorkflow {
       setGlobal(parameters)
       setFiles(newUsedFiles)
     },
+    clearNodes () {
```
Comment (sverhoeven) [inline src/store.ts:171]: Name of function does not cover the functionality, can it be renamed to `clear()` or `clearWorkflow()`.

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Almost there, can you also clear the files.

File: src/store.ts (hunk 2 above)
Comment (rvhonorato) [reply src/store.ts:176]: ops!

**Review by sverhoeven — APPROVED** [review]
Clearing the workflow now works completely.

Thanks for the contribution and feel free to merge it.

**Comment by rvhonorato** [comment]
Thanks for the feedback @sverhoeven, I'll try to help with other issues 😄

**Pull request merged**